src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/MessagePageRenderer.cs

```diff
@@ -0,0 +1,161 @@
+                {
+                    message = UAMessageCenter.Shared().MessageList.Message(messageId);
+                    if (message != null && !message.IsExpired())
+                    {
+                        LoadMessageBody(message);
+                    }
+                    else
+                    {
+                        messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.Unavailable);
+                    }
+                }, () =>
+                {
+                    messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.FetchFailed);
+                });
+            }
+        }
+
+        public void Close()
+        {
+            messagePage.OnRendererClosed(messageId);
+        }
+
+        [Export("webView:decidePolicyForNavigationResponse:decisionHandler:")]
+        public void DecidePolicy(WKWebView webView, WKNavigationResponse navigationResponse, Action<WKNavigationResponsePolicy> decisionHandler)
+        {
+            var response = navigationResponse.Response as NSHttpUrlResponse;
+            if (response.StatusCode >= 400 && response.StatusCode <= 599)
+            {
+                decisionHandler(WKNavigationResponsePolicy.Cancel);
+                if (response.StatusCode == 410)
+                {
+                    messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.Unavailable);
+                }
+                else
+                {
+                    messagePage.OnRendererLoadFailed(messageId, true, MessageFailureStatus.LoadFailed);
+                }
+            }
+            else
+            {
+                decisionHandler(WKNavigationResponsePolicy.Allow);
+            }
+        }
+
+        [Export("webView:didFinishNavigation:")]
+        public void DidFinishNavigation(WKWebView webView, WKNavigation navigation)
+        {
+            messagePage.OnRendererLoaded(messageId);
+        }
+
+        [Export("webView:didFailNavigation:withError:")]
+        public void DidFailNavigation(WKWebView webView, WKNavigation navigation, NSError error)
+        {
+            messagePage.OnRendererLoadFailed(messageId, true, MessageFailureStatus.LoadFailed);
+        }
+
+        [Export("webView:didFailProvisionalNavigation:withError:")]
+        public void DidFailProvisionalNavigation(WKWebView webView, WKNavigation navigation, NSError error)
+        {
+            webView.NavigationDelegate?.DidFailNavigation(webView, navigation, error);
+        }
+
+        protected override void OnElementChanged(VisualElementChangedEventArgs e)
+        {
+            base.OnElementChanged(e);
+
+            if (e.OldElement != null || Element == null)
+            {
+                return;
+            }
+
+            messagePage = e.NewElement as MessagePage;
+            messageId = messagePage.MessageId;
+            messagePage.PropertyChanged += OnElementPropertyChanged;
```

Comment:
For posterity: PageRenderer on iOS doesn't expose an OnElementPropertyChanged method to override, as it does on Android. So we have to make our own and subscribe to the event handler directly. On Android you just override the method.